src/lib.rs

```diff
@@ -355,21 +349,27 @@ pub unsafe extern "C" fn spring_column_text(
     out: *mut c_char,
     out_len: c_int,
 ) -> c_int {
-    let row = &*((*row).0 as *const springql_core::SpringRow);
+    let row = (*row).row();
     let i_col = i_col as usize;
-
-    with_catch(|| springql_core::spring_column_text(row, i_col))
-        .map_or_else(|errno| errno as c_int, |text| strcpy(&text, out, out_len))
+    let result: Result<String, SpringErrno> =
+        with_catch(|| row.get_not_null_by_index(i_col as usize));
+    match result {
+        Ok(v) => {
+            strcpy(&v, out, out_len);
+            v.len() as c_int
```

Comment:
[IMO] Better not to discard the retval from `strcpy()`.

```suggestion
            let v_len = strcpy(&v, out, out_len);
            v_len as c_int
```